src/OpenTelemetry.Collector.Azure/AzureSdkCollector.cs

```diff
@@ -0,0 +1,150 @@
+﻿// <copyright file="AzureSdkCollector.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Collectors.Azure
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using OpenTelemetry.Context;
+    using OpenTelemetry.Trace;
+
+    public class AzureSdkCollector : IDisposable, IObserver<DiagnosticListener>, IObserver<KeyValuePair<string, object>>
+    {
+        private readonly ConcurrentDictionary<Activity, IScope> scopes = new ConcurrentDictionary<Activity, IScope>(new ActivityReferenceEqualityComparer());
+
+        private readonly ITracer tracer;
+
+        private readonly ISampler sampler;
+
+        private List<IDisposable> subscriptions = new List<IDisposable>();
+
+        public AzureSdkCollector(ITracer tracer, ISampler sampler)
+        {
+            this.tracer = tracer;
+            this.sampler = sampler;
+
+            this.subscriptions.Add(DiagnosticListener.AllListeners.Subscribe(this));
+        }
+
+        public void Dispose()
+        {
+            lock (this.subscriptions)
+            {
+                foreach (var subscription in this.subscriptions)
+                {
+                    subscription.Dispose();
+                }
+            }
+        }
+
+        public void OnCompleted()
+        {
+        }
+
+        public void OnError(Exception error)
+        {
+        }
+
+        public void OnNext(KeyValuePair<string, object> value)
+        {
+            if (value.Key.EndsWith("Start"))
```

Comment:
OT philosophy is to throw if things are not right. I.e. you can expect that if e.g. there is a bug (let's say Activity format is not W3C), customer app will stop functioning. 

This is good when a customer calls API themself, but here customer has no control over communication between OT and Azure SDK. 

So I suggest to wrap whole OnNext with try/catch and add TODO to log error.